File: test_tool/api/apimanager.py
import sys

sys.path.append('..')
sys.path.append('../..')

from api.nativecontract import NativeApi
from api.contract import ContractApi
from api.node import NodeApi
from api.restful import RestfulApi
from api.rpc import RPCApi
from api.websocket import WebSocketApi

class APIManager():
	#Singleton
	def __new__(cls):
		if not hasattr(cls, 'instance'):
			cls.instance = super(APIManager, cls).__new__(cls)
		return cls.instance

	def __init__(self):
		self.nodeapi = NodeApi()
		self.contractapi = ContractApi()
		self.nativecontractapi = NativeApi()
		self.restfulapi = RestfulApi()
		self.rpcapi = RPCApi()
		self.websocketapi = WebSocketApi()

	def contract(self):
		return self.contractapi;

	def node(self):
		return self.nodeapi;

	def native(self):
		return self.nativecontractapi;

	def rpc(self):
		return self.rpcapi;

	def restful(self):
		return self.restfulapi;

	def ws(self):
		return self.websocketapi;

API = APIManager()

File: sdk-test-tool/src/main/java/com/ontio/testtool/api/ApiManager.java
package com.ontio.testtool.api;

public class ApiManager {
	static ContractApi contract = null;
	static NodeApi node = null;

	public ContractApi contract() {
		if (contract == null) {
			contract = new ContractApi();
		}
		
		return contract;
	}
	 
	public NodeApi node() {
		if (node == null) {
			node = new NodeApi();
		}
		
		return node;
	}
}
